frontend/src/components/HeaderUser.tsx

```diff
@@ -0,0 +1,58 @@
+import "../index.css";
+import React, { useState } from "react";
+import { SNOWPACK_PUBLIC_API_URL } from "../model";
+
+const logout = async () => {
+  await fetch(`${SNOWPACK_PUBLIC_API_URL}/api/logout`, {
+    method: "POST",
+  })
+    .then((res) => {
+      if (res.ok) {
+        return true;
+      } else {
+        throw new Error("Could not log out.");
+      }
+    })
+    .catch((error) => {
+      console.log(error);
+      return false;
+    });
+  // Redirect to login page
+  window.location.href = "/";
+};
+
+const showSettings = () => {
+  return (
+    <div class="settings-list" aria-labelledby="dropdownMenuLink">
+      <a class="dropdown-item" onClick={logout}>
+        Log out
+      </a>
+    </div>
+  );
+};
+
+export const HeaderUser = ({ username }: { username: string }) => {
+  const [userSettings, setUserSettings] = useState<boolean>(false);
+
+  const showSettingsClickHandle = () => {
+    setUserSettings(!userSettings);
+  };
+
+  return (
+    <div className="d-flex justify-content-end header-user">
+      <div className="p-2">
+        <a
+          class="btn basic-button dropdown-toggle"
+          id="dropdown-button"
```

Comment:
If you change the css file according to my suggestion, you can remove the id entirely here.